### ISSUE TITLE
Implementations of IList create warning related to the indexer

### ISSUE THREAD

**Issue by tom-englert** [EWSoftware/SHFB#22]
All my classes implementing System.Collections.IList produce this kind of warning:

Warning	23	BuildAssembler : warning : ResolveReferenceLinksComponent: [P:TomsToolbox.ObservableCollections.ObservableListAdapter`1.Item(System.Int32)] MSDN URL not found for target 'P:System.Collections.IList.Item(System.Int32,System.Object)'.

Why is it looking for System.Collections.IList.Item(System.Int32,System.Object) instead of System.Collections.IList.Item(System.Int32)?

Is there a setting or plugin that can fix this?

Im using VS2013, C#, SHFB 2015.1.12.0. The source related to the issues can be found at https://tomstoolbox.codeplex.com/SourceControl/latest

**Comment by EWSoftware** [comment]
I built your code with the latest release and didn't see those warnings.  I suspect that this is related to a bug fixed in v2014.11.22.0 related to some invalid IDs in the core framework comments files and the MSDN content service.  You should be able to resolve it by deleting the content ID cache folders and files (%LOCALAPPDATA%\EWSoftware\Sandcastle Help File Builder\Cache).  After deleting the caches, the next build will recreate the cache files and any cached invalid IDs will be replaced with good ones.

**Comment by tom-englert** [comment]
Deleting the cache fixed the problem. 
Thanks.

**Comment by tom-englert** [comment]
Deleting the cache just made the first build so slow that I did not realize the build was not yet finished!
The issue is still there, I get it on two different machines.
The help files will build in release configuration only, maybe you tried the debug build?

**Comment by EWSoftware** [comment]
I downloaded your latest code and built it.  Debug or Release it made no difference and the help file built in each one.  I also deleted my cache folder and let it rebuild it and still saw no warning.

Did you install SHFB using the guided installer or did you download the code and build it and the reflection data?  If the later perhaps your reflection data isn't the same.  If that is the case, removing the version you built and using the guided installer to install the tools and reflection data in the usual manner may resolve the issue.  I don't see that member ID referenced in your XML comments so I don't know where it's coming from.

**Comment by tom-englert** [comment]
Finally I found the differences of our systems: My project is using Microsoft's Code Contracts (https://visualstudiogallery.msdn.microsoft.com/1ec7db13-3363-46c9-851f-1ce455f66970), which is adding contracts information to the XML comments. Without having installed code contracts you wont get this extra info. 

Here is a link to a zip file containing both XML with and without contracts decorations:
http://1drv.ms/1vExbH2

and this is the output generated: 
http://tomstoolbox.azurewebsites.net/html/4B6FEE5C.htm

**Comment by EWSoftware** [comment]
I'll install the package and see if there's something in SHFB that's causing it or if it can be worked around.  If the contracts tool is introducing invalid links into the comments files, it may be something that needs to be reported to the owner of that project.  I'll let you know what I find.

**Comment by EWSoftware** [comment]
I was able to find the problem.  The getter and setter IDs are in the form of method calls.  The transformations are stripping the "get_" and "set_" prefixes but for the setter, it also needs to strip the last parameter which is the type being assigned to the property.  I've fixed the transformations so that they do that and the warnings are gone.  It'll be in the next release.  No harm done for the time being.  It just won't render clickable links for the setter contract section on those topics.
